src/ae/contract.ts

```diff
@@ -0,0 +1,212 @@
+/*
+ * ISC License (ISC)
+ * Copyright (c) 2022 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * Contract module - routines to interact with the æternity contract
+ *
+ * High level documentation of the contracts are available at
+ * https://github.com/aeternity/protocol/tree/master/contracts and
+ *
+ * @module @aeternity/aepp-sdk/es/ae/contract
+ * @export Contract
+ * @example import { Contract } from '@aeternity/aepp-sdk'
+ */
+
+// @ts-expect-error TODO remove
+import Ae from '.'
+import ContractCompilerHttp from '../contract/compiler'
+import getContractInstance from '../contract/aci'
+// @ts-expect-error TODO remove
+import { AMOUNT, MIN_GAS_PRICE } from '../tx/builder/schema'
+// @ts-expect-error TODO remove
+import { decode, produceNameId } from '../tx/builder/helpers'
+// @ts-expect-error TODO remove
+import { getNetworkId } from '../node'
+
+export abstract class _Contract {
+  /**
+   * Utility method to create a delegate signature for a contract
+   * @alias module:@aeternity/aepp-sdk/es/ae/contract
+   * @category async
+   * @param ids The list of id's to prepend
+   * @param opt Options
+   * @return Signature in hex representation
+   */
+  async delegateSignatureCommon (
+    ids: string[] = [],
+    opt: { onAccount?: string | object } = {}): Promise<string> {
```

Comment:
`onAccount` is accepted by one of the function we are passing options to, don't define options by yourself, combine options types of `getNetworkId` and `sign` instead